SingularityUI/app/actions/api/base.es6

```diff
@@ -46,7 +46,15 @@ export function buildApiAction(actionName, opts={}) {
 
       let options = optsFunc(...args);
       return fetch(config.apiRoot + options.url, _.extend({credentials: 'include'}, _.omit(options, 'url')))
-        .then(response => response.json())
+        .then(response => {
+          if (response.headers.get('Content-Type') === 'application/json') {
+            return response.json();
+          } else {
+            // All API calls respond with JSON, so if we're not getting JSON it's bad
```

Comment:
Not sure if I completely agree with this... I like how you're leveraging the content type to decide how to decode the response, but we should really be keying off the response's status code to decide if there was an error or not